src/main/java/edu/ksu/canvas/attendance/services/AttendanceSectionService.java

```diff
@@ -25,9 +34,110 @@ public AttendanceSection getFirstSectionOfCourse(long canvasCourseId) {
         return sections.isEmpty() ? null : sections.get(0);
     }
 
+    public List<AttendanceSection> getSectionByCanvasCourseId(long canvasCourseId) {
+        return sectionRepository.findByCanvasCourseId(canvasCourseId);
+    }
 
     public List<AttendanceSection> getSectionsByCourse(long canvasCourseId) {
         return sectionRepository.findByCanvasCourseId(canvasCourseId);
     }
 
+    /**
+     * @throws RuntimeException when courseForm is null
+     */
+    public void save(CourseConfigurationForm courseForm, long canvasCourseId) {
+        Validate.notNull(courseForm, "courseForm must not be null");
+
+        List<AttendanceSection> sections = sectionRepository.findByCanvasCourseId(canvasCourseId);
+        if(sections == null || sections.isEmpty()) {
+            RuntimeException e = new IllegalArgumentException("Cannot load data into courseForm for non-existant course");
```

Comment:
IllegalArgumentException is probably not applicable here. IllegalArgumentException is used to indicate that an invalid argument was passed. In this case the argument could be valid but the data is wrong.

Generally IllegalArgumentExceptions indicate programmer error or a bug. I try to think of them as a super poor man's unit test. They should (hopefully) not be thrown in production.